### ISSUE TITLE
[$175] Need to optimize UI for showing join/decline button on listing page 

### ISSUE THREAD

**Issue by RishiRajSahu** [appirio-tech/connect-app#3407]
We show join/decline button for the invited users on listing page, which takes lot of space on the project bar and even after accepting/decline the invite the space still stays occupied for those buttons. 
We need to get some new optimized UI designs for those buttons.

<img width="1183" alt="Screen Shot 2019-12-11 at 9 48 20 AM" src="https://user-images.githubusercontent.com/11248478/70591421-1b2f1380-1bfc-11ea-9175-b8b8a87d9251.png">
 
fyi @acshields @vikasrohit

**Comment by mahestro** [comment]
Hi @RishiRajSahu. We can take care of this, it looks quite simple. I have a couple doubts before moving forward:

1. What should happen after the user clicks `accept`?
2. What should happen after the user clicks `decline`?

Before you reply, I wouldn't dare to come to you without a proposal. I might be

**Comment by mahestro** [comment]
Additionally, question #3: is there a dev environment available online I can access to quickly test the features?

**Comment by RishiRajSahu** [comment]
@mahestro those options simply disappears, but the space is still reserved in the UI where the buttons were, check the empty section I have highlighted in the screenshot above.

Yeah there is a dev env of Connect https://connect.topcoder-dev.com/ , that's a good idea why don't you see this actually in action . I have invited you to couple of projects in dev env, you can login and see how the buttons reacts when doing either of the action.

Try switching the view of projects listing page, the screenshot above shows list view, try switching to card view as well and see the behaviour.

**Comment by mahestro** [comment]
Thanks for the feedback. If there's an empty space after the action happens, there is room for improvement. I think the user would benefit from having some feedback that indicates what happened after she clicked any of the buttons: did I get assigned? did the process work? 

I tried logging into the dev environment but it's not working using my prod credentials. Is there anything I'm missing? 🤔

**Comment by RishiRajSahu** [comment]
@mahestro yep credentials will be different.

**Comment by mahestro** [comment]
Just tried the behavior, I get the idea. Thanks.

For the buttons design, should we stay within the current behavior? I know maybe for speed purposes it's acceptable, however, if I may suggest, I think at least we should display the popup notificatoin from bottom left when the user takes action (success or failure). Would that be ok?

**Comment by RishiRajSahu** [comment]
@mahestro changing the buttons design is the main things, buttons are something that's taking space which we want to remove.
Current behaviour - 
1. when user clicks either of button the loading spinner appears
2. Once the operation is done we are already showing notification on left bottom, so we are good there.

So, If I say what's more important is just to reduce the space occupied by the buttons, maybe using some smaller user actionable icons.

**Comment by mahestro** [comment]
Understood! Thanks for the clarification. We're working on the update. We should be having something to see by the 27th or earlier.

**Comment by mahestro** [comment]
Hi @RishiRajSahu. 

We have the first iteration of the design using the connect branding and optimizing the space. Let me know if that's something that would work or if you have any feedback.

https://marvelapp.com/a35h37b

**Comment by RishiRajSahu** [comment]
@mahestro added a comment there on marvelapp.

**Comment by mahestro** [comment]
Thanks for the great feedback @RishiRajSahu 

If design looks good, here are the source files in Abstract:
- List view: https://share.goabstract.com/7016973b-b02e-4c7c-a3b4-9874c093d817?mode=design&sha=785ab46f48a70793beec2e4ff6943dbf362192b7
- https://share.goabstract.com/f3fc2050-5e53-4832-88d0-04fbbcdc54d6?mode=design&sha=785ab46f48a70793beec2e4ff6943dbf362192b7

Btw, how do you prefer to receive them in terms of handoff? Are you ok with the sketch source files or maybe you are more comfortable with some outcomes like Zeplin does?

**Comment by RishiRajSahu** [comment]
@maxceem adding this for CF

**Comment by maxceem** [comment]
@RishiRajSahu I guess so, but I don't have access to the designs to check them out.

Is it possible to get them as `Marvel Handoff`? It would be the best way to share them with the members.

**Comment by RishiRajSahu** [comment]
@Luis Millán <mahestro@topcoder.com> Could we get the marvel handoff for
the designs ?

**Comment by RishiRajSahu** [comment]
@mahestro Could we get the marvel handoff for the designs ?

**Comment by mahestro** [comment]
Hi guys, handoff is already enabled in the marvelapp project (same one).

https://marvelapp.com/a35h37b/screen/65065855/handoff

**Comment by maxceem** [comment]
Great, thanks @mahestro.

**Comment by maxceem** [comment]
@RishiRajSahu yes, this is good for CF. I would like to confirm only one thing. So far for consistency in Connect App we use buttons in the current style with rectangular buttons.

So for the new design, I guess we have to make the next adjustments:

![image](https://user-images.githubusercontent.com/146016/72400672-ff7fd580-3784-11ea-9e70-34f5e265e051.png)

- instead of green round button use our current blue rectangular button
- instead of red round button use a red rectangular button
   
   ![image](https://user-images.githubusercontent.com/146016/72400983-c98f2100-3785-11ea-947f-4d071dece66a.png)

- the "you are invited" button we can implement as per new design, and just replace the green color with our current main blue color `$tc-dark-blue-100`.

**Comment by RishiRajSahu** [comment]
@maxceem agree but we will miss the tick sign from the rounded button, is it going way out of them keeping the rounded button with our colours ?

**Comment by maxceem** [comment]
>  but we will miss the tick sign from the rounded button

I think we can still add a "tick" inside a blue rectangular button too. We can use this icon https://github.com/appirio-tech/connect-app/blob/dev/src/assets/icons/ui-check.svg and style it with CSS according to design if necessary.

![image](https://user-images.githubusercontent.com/146016/72495986-2ad8f200-3863-11ea-8e6f-3903a3d07b64.png)

> is it going way out of theme keeping the rounded button with our colours 

The biggest downside, is that we would have to create one more style for a button. When later we would switch to a new theme we would have to take care of this case separately, while if we reuse existent buttons, we could switch to a new design by updating button style in one place.

> Also, 'you're invited' button is also round so think we could go with round buttons and current join/decline button is also round

We don't have a UI element for such a place yet. So we can implement it in any style. We can also make it `rectangular` for now.

@RishiRajSahu let me know which style would you prefer to use at the end `rectangular` or `round`?

**Comment by RishiRajSahu** [comment]
@maxceem let's keep it round, I think later if we would be switching to new theme it would anyway going to be big rework so we can bear that effort and we can even discard the old styles(both rectangular and rounded) and use a single new one(whatever it would be). Moreover this design is reviewed by product manager so don't want to alter it now, but still thanks for pointing out.

**Comment by maxceem** [comment]
Got it @RishiRajSahu. So we implement new round buttons and the only things, we replace a new `green` color with our main `blue` color `$tc-dark-blue-100`, right?

**Comment by maxceem** [comment]
Sum up:
- currently, if a user is invited to some project we are showing buttons to decline/accept an invitation on the project listing page like this:

    - card view:

        ![image](https://user-images.githubusercontent.com/146016/72703255-157c0480-3b90-11ea-9b90-ae03fc600eb1.png)

   - grid view:

        ![image](https://user-images.githubusercontent.com/146016/72703298-2e84b580-3b90-11ea-9a49-9f3da5881746.png)

   - see demo page using use `pshah_customer` https://connect.topcoder-dev.com/projects?sort=updatedAt%20desc

- To keep the space in the `grid view` we want to update UI to next one:

   - grid view: https://marvelapp.com/a35h37b/screen/65065857/handoff
     - Note: that we should not have an additional column anymore for these buttons.
   - card view: https://marvelapp.com/a35h37b/screen/65065855/handoff

- Implement the new UI as per provided design with one exception, instead of new green color use our current blue color `$tc-dark-blue-100`.

- Implement a reusable component for this purpose so it can be used in two places grid and card view:

   ![image](https://user-images.githubusercontent.com/146016/72703583-e4500400-3b90-11ea-97fd-7e167874fefd.png)

   - now we implemented these buttons 2 times for each of list view which is not good

- when we click any of the buttons, the loader indicator should be shown instead of buttons, the same as it's done now:

   ![image](https://user-images.githubusercontent.com/146016/72703700-31cc7100-3b91-11ea-86a6-1159eb7013a1.png)

- for testing use user `pshah_manger` to invite user `pshah_customer` and see the project listing page as a `pshah_customer` 

- we have icons already in the repo:
   - https://github.com/appirio-tech/connect-app/blob/dev/src/assets/icons/ui-check.svg
   - https://github.com/appirio-tech/connect-app/blob/dev/src/assets/icons/arrow-9px-carret-down-normal.svg
   - you may use CSS to style them appropriatly

#### General requirements:
- For styling use ONLY colors from the https://github.com/appirio-tech/tc-ui/blob/feature/connectv2/src/styles/_variables.scss. If some new color is introduced in design, please replace it with the closest one from the variables. The exception is for a new green color, replace it with the blue color $tc-dark-blue-100. If you have any concerns regarding the colors, please, ask on the forum.

- Avoid using multilevel nesting in SCSS. As we use CSS Modules we don’t have to add prefixes or nest class selectors.

- Don’t use `:global` in CSS Modules unless you have to change some global styles. New styles shouldn’t use `:global`.

**Comment by ChanKamWo** [comment]
@bug-bash-helper assign

**Comment by maxceem** [comment]
@ChanKamWo

As no PR has been provided for 12 hours this issue has been unassigned and now is open for pickup.

**Comment by rashmi73** [comment]
@bug-bash-helper assign

**Comment by rashmi73** [comment]
@maxceem working it may take more than 12 hours as bit lengthy

**Comment by maxceem** [comment]
@rashmi73 just in case would like to let you know that this issue could be held no more than 36 hours in total. There are still 13 hours to complete the issue.

**Comment by maxceem** [comment]
@rashmi73 As PR hasn't been raised for this issue more than 36 hours, it was unassigned and now is open for pick up.

Note, maximum time to solve this issue is 36 hours.

**Comment by PrakashDurlabhji** [comment]
@bug-bash-helper assign

**Comment by PrakashDurlabhji** [comment]
@maxceem update you shortly

**Comment by PrakashDurlabhji** [comment]
@maxceem in some time

**Comment by maxceem** [comment]
@PrakashDurlabhji as this issue has not been solved for 36 hours it is unassigned and now is open for pick up.

- The prize for the issue raised to $175
- Maximum time to solve this issue is <b>24</b> hours.

**Comment by rashmi73** [comment]
@bug-bash-helper assign

**Comment by maxceem** [comment]
@rashmi73 as you've already worked on this issue before and didn't provide a solution for 36 hours, please, don't pick up this issue and let other people try.

**Comment by rashmi73** [comment]
@maxceem as I already worked on it so I m midway and complete faster, can I take this issue if i give up my other issue?

**Comment by maxceem** [comment]
@rashmi73 could you please create a Pull Request with what you have done so far for this issue, I'll have a look if there is big enough work already done to continue.

**Comment by yoution** [comment]
@bug-bash-helper assign

**Comment by yoution** [comment]
@bug-bash-helper https://github.com/appirio-tech/connect-app/pull/3518 is ready for review

**Comment by maxceem** [comment]
@yoution generally works great, there are few issues to fix, please have a look at review https://github.com/appirio-tech/connect-app/pull/3518#pullrequestreview-348436165

**Comment by vikasrohit** [comment]
After this change it seems like we are now overlapping the customer name with next column, created separate issue #3525  for it.

**Comment by nithyaasworld** [comment]
@maxceem Verified in dev. There is no functional issues.

UX Feedback:
1. On clicking 'Decline' button, it declines the project immediately. If user presses the button accidentally, then he needs to be invited again. If possible, would like a 'Decline confirmation message' with 'yes or no' options like suggested here: https://github.com/appirio-tech/connect-app/issues/3407#issuecomment-567962801

UI Issues:
2. Tick mark in the 'Join button' is visible only for the first invitation when there are multiple invitations 

![image](https://user-images.githubusercontent.com/5114190/73313053-30b1d880-4250-11ea-819c-3b7638e147cf.png)
![image](https://user-images.githubusercontent.com/5114190/73313061-36a7b980-4250-11ea-994c-9531d143e624.png)

3. Location and look of the 'You're invited' button can be improved to add elegance

![image](https://user-images.githubusercontent.com/5114190/73314626-53de8700-4254-11ea-907f-e952804fe44d.png)

4. CSS of all the new buttons should be updated to match with the design handoff. Example below:
![image](https://user-images.githubusercontent.com/5114190/73315682-bcc6fe80-4256-11ea-8ff5-7cbd79a23f44.png)

**Comment by maxceem** [comment]
@nithyaasworld Visual Issues 2, 3, 4, 5 are fixed and ready for review on DEV.

@RishiRajSahu what do you think regarding 1?
> UX Feedback:
> 🔹1. On clicking 'Decline' button, it declines the project immediately. If user presses the button accidentally, then he needs to be invited again. If possible, would like a 'Decline confirmation message' with 'yes or no' options like suggested here: #3407 (comment)

**Comment by nithyaasworld** [comment]
Thanks @maxceem 

Just 2 more minor UI issues:
1. We still need to match the CSS of the Join and Decline buttons as well as the CSS of the text inside both these buttons with the design-handoff
2. 'You're invited' button in the list view, has to be vertically in line with the Manager-avatar icons and the column-header 'Managers'
![image](https://user-images.githubusercontent.com/5114190/73409404-2f061480-4325-11ea-99fb-d740f9803ebf.png)

**Comment by maxceem** [comment]
@nithyaasworld thanks for pointing out. Fixed 2 more issues from above https://github.com/appirio-tech/connect-app/issues/3407#issuecomment-580027595.

Ready for testing on DEV.

**Comment by nithyaasworld** [comment]
Thanks @maxceem 

Verified No:6.

But for 4, we still need to match the CSS of the **Join and Decline** button **text.** 
![image](https://user-images.githubusercontent.com/5114190/73532530-e04e9c80-4442-11ea-8618-f46110136f4f.png)

**Comment by maxceem** [comment]
Ah, sorry @nithyaasworld, this time it must much perfectly, please have a look.

**Comment by RishiRajSahu** [comment]
@maxceem I think we don't have designs yet for Yes/No option before actually declining the invite ? Do you think we can use our standard popup alert for this ?

**Comment by maxceem** [comment]
@RishiRajSahu right, we don't have a new design for it. And we can use the same confirmation dialog we have when we are trying to delete a topic/post:

![image](https://user-images.githubusercontent.com/146016/73806988-f58e4700-4805-11ea-9afe-ec5a67f145ce.png)

**Comment by nithyaasworld** [comment]
Thanks @maxceem Verified in Dev.

**Comment by RishiRajSahu** [comment]
@maxceem looks good to me, Let's create a separate git ticket to make this change. and let's use this text **"**Are you sure you want to decline the invite?**"** with buttons saying `Cance` and `Decline Invite`

**Comment by maxceem** [comment]
Created a follow-up issue https://github.com/appirio-tech/connect-app/issues/3697

**Comment by vikasrohit** [comment]
Closing the issue for cleanup. Please feel free to create separate regression issues.
